### PR TITLE
Improve h2d.Interactive collision with shape.

### PR DESCRIPTION
**Pull request by Yanrishatum**
* Ignore width/height if `shape` is set.
* Add invDet cache to h2d.Interactive
* Simplify `eventToLocal`
* Disable `isEllipse` for shapes.
* Some documentation.
* Visibility check was relocated before collision check.

### CODE REVIEW

File: h2d/Scene.hx [hunk 1]
```diff
@@ -289,13 +264,48 @@ class Scene extends Layers implements h3d.IDrawable implements hxd.SceneEvents.I
 			}
 			if( !visible ) continue;
```
Comment (ncannasse) [inline h2d/Scene.hx]: I would like to keep width/height testing before visibility testing, in case there are many Interactive within deep hierarchies.

Comment (Yanrishatum) [reply h2d/Scene.hx]: And for case of shape interactive? Should it check for visibility before or after shape collision check?

**Comment by Yanrishatum** [comment]
Oh, and another thing. Maybe `getBounds` should actually get shape bounds if it's present instead of `width` / `height` ?

File: h2d/Scene.hx (hunk 1 above)
Comment (ncannasse) [reply h2d/Scene.hx]: I guess we should be uniform here and check after in all cases

**Comment by ncannasse** [comment]
Regarding getBounds it's a bit tricky.
setting backgroundColor will not work with shape (would require to be able to draw that shape)
and for "forSize", it's only when the interactive is inside a Flow, and usually you have an additional graphics that goes with it.

**Comment by ncannasse** [comment]
Thanks!

**Pull request merged**